src/libraries/System.Net.Sockets/tests/FunctionalTests/CreateSocketTests.cs

```diff
@@ -445,6 +446,25 @@ public async Task Ctor_SafeHandle_Listening_Success(bool shareSafeHandle)
             }
         }
 
+        [DllImport("libc")]
+        static extern int socket(int domain, int type, int protocol);
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Linux)]
+        public void Ctor_SafeHandle_UnknownSocket_Success()
+        {
+            const int PF_NETLINK = 16;
+            const int NETLINK_ROUTE = 0;
+            const int SOCK_DGRAM = 2;
+
+            int fd = socket(PF_NETLINK, SOCK_DGRAM, NETLINK_ROUTE);
+            Assert.True(fd >= 0);
+            using (Socket netlink = new Socket(new SafeSocketHandle((IntPtr)fd, ownsHandle: true)))
+            {
+                Assert.Equal(AddressFamily.Unknown, netlink.AddressFamily);
```

Comment:
I have code to dump routing table - essentially equivalent of `netstat -r`. I can probably simplify it and use as test if you think that would be appropriate. I was planning to post that eventually as example to the issues asking for Netlink support.

Reply:
Yeah, I think something would be useful, just to validate that we're actually able to interact with such a socket.

Reply:
updated. This now sends message to kernel (to get routes) and verifies valid response (without parsing route details) 

I had to do few more touches to get Bind() working. I created new EndPoint type but it seems like Socket code use some internal implementation I was not able to override. With that, it would go to PAL anyway and fail. I don't know if there is better way to do this or what is the reason for duplicating addresses.